lib/spack/spack/fetch_strategy.py

```diff
@@ -917,6 +931,80 @@ def clone(self, dest=None, commit=None, branch=None, tag=None, bare=False):
                     git(*pull_args, ignore_errors=1)
                     git(*co_args)
 
+    def _sparse_clone_src(self, commit=None, branch=None, tag=None, **kwargs):
+        """
+        Use git's sparse checkout feature to clone portions of a git repository
+
+        Arguments:
+            commit (str or None): A commit to fetch from the remote. Only one of
+                commit, branch, and tag may be non-None.
+            branch (str or None): A branch to fetch from the remote.
+            tag (str or None): A tag to fetch from the remote.
+        """
+        dest = self.stage.source_path
+        git = self.git
+
+        if self.git_version < spack.version.Version("2.25.0.0"):
+            tty.warn(
+                (
+                    f"{self.package.name} is configured for git sparse-checkout "
+                    "but the git version is too old to support sparse cloning. "
+                    "Cloning the full repository instead."
+                )
+            )
+            self.clone_src(commit, branch, tag)
+        else:
+            # default to depth=2 to allow for retention of some git properties
+            depth = kwargs.get("depth", 2)
```

Comment:
Not necessarily for this PR, but do you believe we should be applying this everywhere through `_clone_src` as well?

Reply:
🤷🏻  I don't have strong feelings. I forgot I put this here. I think it was your suggestion from slack... 😜

Reply:
Yeah, let's plan for future work to apply it everywhere unless someone disagrees.